extensions/tokenizers/rust/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern crate tokenizers as tk;
+
+use std::str::FromStr;
+use tk::tokenizer::{EncodeInput, Encoding};
+use tk::FromPretrainedParameters;
+use tk::Tokenizer;
+
+use jni::objects::{JObject, JString};
+use jni::sys::{jboolean, jlong, jlongArray, jobjectArray, jsize, JNI_TRUE};
+use jni::JNIEnv;
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_createTokenizer(
+    env: JNIEnv,
+    _: JObject,
+    input: JString,
+) -> jlong {
+    let identifier: String = env
+        .get_string(input)
+        .expect("Couldn't get java string!")
```

Comment:
AFAIK, `.expect` will trigger a Rust panic and default crash the VM. Alternatively, the code could check for the erroneous case and use `env.throw` for a safer erroneous result handling.

Reply:
Maybe we should throw a OutOfMemoryError here. I will try to improve it in a separate PR.